torch/_inductor/compile_fx_async.py

```diff
@@ -204,9 +206,8 @@ def __init__(
     ) -> None:
         self._fast_output_code = fast_output_code
         self._optimized_output_code = None
-        self._progression_futures = progression_futures
+        self._progression_futures = list(progression_futures)
```

Comment:
I really wish this weren’t necessary, but mypy is right to complain since we’re mutating the list by inserting Nones, which means anyone holding a reference to it now sees a List[Optional[...]], potentially violating their expectations.

Reply:
Maybe change the type of progression_futures to a `Sequence[Future[...]]` so the caller knows it won't be mutated...